TeachingRecordSystem/tests/TeachingRecordSystem.Core.Tests/Services/TrsDataSync/TrsDataSyncHelperTests.Alert.cs

```diff
@@ -45,11 +47,287 @@ await DbFixture.WithDbContext(async dbContext =>
         });
     }
 
+    [Fact]
+    public async Task SyncAlert_WithDeactivatedEvent_SetsDeletedOnAttribute()
+    {
+        // Arrange
+        var person = await TestData.CreatePerson(b => b.WithSyncOverride(false));
+        var qualificationId = Guid.NewGuid();
+        var auditDetailCollection = new AuditDetailCollection();
+        var entity = await CreateNewAlertEntityVersion(qualificationId, person.ContactId, auditDetailCollection);
+
+        Clock.Advance();
+        var deactivatedVersion = await CreateDeactivatedEntityVersion(entity, auditDetailCollection);
+
+        // Act
+        await Helper.SyncAlert(deactivatedVersion, auditDetailCollection, ignoreInvalid: false, createMigratedEvent: true);
+
+        // Assert
+        await DbFixture.WithDbContext(async dbContext =>
+        {
+            var mq = await dbContext.Alerts.IgnoreQueryFilters().SingleOrDefaultAsync(p => p.DqtSanctionId == qualificationId);
+            Assert.NotNull(mq);
+            Assert.Equal(Clock.UtcNow, mq.DeletedOn);
+        });
+    }
+
+    [Fact]
+    public async Task SyncAlert_WithDqtCreateAudit_CreatesExpectedEvents()
+    {
+        // Arrange
+        var person = await TestData.CreatePerson();
+        var qualificationId = Guid.NewGuid();
+        var auditDetailCollection = new AuditDetailCollection();
+        var initialVersion = await CreateNewAlertEntityVersion(qualificationId, person.ContactId, auditDetailCollection, addCreateAudit: true);
+
+        // Act
+        await Helper.SyncAlert(initialVersion, auditDetailCollection, ignoreInvalid: false, createMigratedEvent: true);
+
+        // Assert
+        var events = await GetEventsForAlert(qualificationId);
+
+        await Assert.CollectionAsync(
+            events,
+            async e =>
+            {
+                var createdEvent = Assert.IsType<AlertCreatedEvent>(e);
```

Comment:
Perhaps this and the following event need to have the order swapped in this collection assertion (i.e. to stop the test failing)?

Reply:
Test flakiness seems to be down to cached events in the DB; I think I've fixed that now.